clippy_lints/src/formatting.rs

```diff
@@ -127,6 +149,28 @@ fn check_else_if(cx: &EarlyContext, expr: &ast::Expr) {
     }
 }
 
+/// Implementation of the `POSSIBLE_MISSING_COMMA_IN_ARRAY_FORMATTING` lint for array
+fn check_array(cx: &EarlyContext, expr: &ast::Expr) {
+    if let ast::ExprKind::Array(ref array) = expr.node {
+        for element in array {
+            if let ast::ExprKind::Binary(ref op, ref lhs, _) = element.node {
+                let space_span = mk_sp(lhs.span.hi, op.span.lo);
```

Comment:
Not actually checked but something like

```rust
macro_rules! foo {
    () => { 42 }
}

let _ = [
    foo!()
    -4
];
```
might fail because the `42` and the `4` have different context.